### PR TITLE
Improve folder enumeration speed

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
For #4180

**Details of Changes**
- Removes caching feature
- Perf tweaks to folder loading code: removed calls to StorageFolder methods when not strictly needed
- Fixes an issue for which icons are not loaded when browsing mtp devices

**Warning**
These changes creates issues with the "current" details view, I find the app crashing every 5 folders.
This only happens with the WCT datagrid layout mode, the "new" details view works good.
This actually looks like a bug with WCT and has been known for a while (#4304) but these changes seem to make the crash more common.
Depending on how we plan to ship the next version (new or old details view) it may make sense to wait before doing this change, but I'd love to get some feedbacks on how well this PR work for you.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested loading folders from disk, network, mtp (phone)

### CODE REVIEW

File: Files/ViewModels/ItemViewModel.cs [hunk 1]
```diff
@@ -740,17 +744,16 @@ public async Task LoadExtendedItemProperties(ListedItem item, uint thumbnailSize
...
                                     using var Thumbnail = await matchingStorageFile.GetThumbnailAsync(ThumbnailMode.SingleItem, thumbnailSize, ThumbnailOptions.UseCurrentScale);
                                     using var headerThumbnail = loadGroupHeaderInfo && isFileTypeGroupMode ? await matchingStorageFile.GetThumbnailAsync(ThumbnailMode.DocumentsView, 36, ThumbnailOptions.UseCurrentScale) : null;
                                     if (Thumbnail != null)
                                     {
                                         await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(async () =>
                                         {
-                                            item.FileImage = new BitmapImage();
                                             item.CustomIconData = await Thumbnail.ToByteArrayAsync();
-                                            await item.FileImage.SetSourceAsync(Thumbnail);
+                                            item.FileImage = await item.CustomIconData.ToBitmapAsync();
```
Comment (duke7553) [inline Files/ViewModels/ItemViewModel.cs:756]: If there isn't any IconData, then why reference it here?

**Comment by yaichenbaum** [comment]
> Depending on how we plan to ship the next version (new or old details view) it may make sense to wait before doing this change, but I'd love to get some feedbacks on how well this PR work for you.

@gave92 The new details layout will replace the old one in v2.

File: Files/ViewModels/ItemViewModel.cs (hunk 1 above)
Comment (gave92) [reply Files/ViewModels/ItemViewModel.cs:756]: Here `fileIconInfo.IconData` is null (the icon could not be loaded from FTP) but `item.CustomIconData` (icon from StorageFile's thumbnail) is not.

The previous code did not work:
```cs
item.CustomIconData = await Thumbnail.ToByteArrayAsync(); --> this reads the Thumbnail stream to the end
await item.FileImage.SetSourceAsync(Thumbnail);  --> this is left with a stream already read to the end (empty)
```

Comment (duke7553) [reply Files/ViewModels/ItemViewModel.cs:756]: 👍

**Review by lukeblevins — DISMISSED** [review]
This looks like a good improvement as long as we remove the old details layout before v2

**Comment by yaichenbaum** [comment]
@gave92 It seems like the status bar isn't updating anymore. Another issue is that everything seems very choppy when loading large directories.

**Comment by gave92** [comment]
@yair100 is status bar working in `main` for you?

**Comment by yair100** [comment]
> @yair100 is status bar working in `main` for you?

It looks like it's an issue in `main` as well.

**Comment by gave92** [comment]
>  Another issue is that everything seems very choppy when loading large directories.

@yair100 is this still an issue? If so could you clarify? I would not expect it to be "more choppy" than the current.

**Comment by yair100** [comment]
> > Another issue is that everything seems very choppy when loading large directories.
> 
> @yair100 is this still an issue? If so could you clarify? I would not expect it to be "more choppy" than the current.

It appears to be working better now, I noticed a different issue where the app seems to scroll down after loading the system 32 directory in the details layout.

**Comment by gave92** [comment]
> It appears to be working better now, I noticed a different issue where the app seems to scroll down after loading the system 32 directory in the details layout.

Could you get me a screen recording of this?

**Comment by yaichenbaum** [comment]
> > It appears to be working better now, I noticed a different issue where the app seems to scroll down after loading the system 32 directory in the details layout.
> 
> Could you get me a screen recording of this? I have trouble reproducing

I reset Files and I can't reproduce it anymore 😂

**Review by yair100 — APPROVED** [review]
LGTM

**Pull request merged**